projects/create-and-vary-a-licence-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/LicenceConditionApplier.kt

```diff
@@ -65,13 +65,42 @@ class LicenceConditionApplier(
                 )
             }
 
-            else -> listOf(
-                ActionResult.Ignored(
-                    "Multiple Custodial Sentences",
-                    properties
-                )
+            else -> associateLicenceToLongestSentence(sentences, com, activatedLicence, occurredAt, properties)
+        }
+    }
+
+    fun associateLicenceToLongestSentence(
+        sentences: List<Custody>,
+        com: PersonManager,
+        activatedLicence: ActivatedLicence,
+        occurredAt: ZonedDateTime,
+        properties: Map<String, String>
+    ): List<ActionResult> {
+        val longestSentenceByEnteredDate = sentences
+            .filter { it.disposal.enteredSentenceEndDate != null }
+            .maxByOrNull { it.disposal.enteredSentenceEndDate!! }
+
+        longestSentenceByEnteredDate?.let {
+            return applyLicenceConditions(
+                SentencedCase(
+                    com,
+                    it.disposal,
+                    licenceConditionService.findByDisposalId(it.disposal.id)
+                ),
+                activatedLicence,
+                occurredAt
             )
         }
+
+        //calculated end date should always be populated
+        sentences
+            .maxBy { it.disposal.endDate }.let {
+                return applyLicenceConditions(
```

Comment:
Is this potentially adding the licence conditions to two of the events?  i.e. the one with the latest `enteredSenetenceEndDate`, and then the one with the latest `endDate`?

Reply:
it should not to, as line 84 should return if the condition is met.